Add Hugo based web site.

sophokles73 opened the pull request:
I have added folder `site/` which contains the source code for a Californium website to be hosted at https://www.eclipse.org/californium`. The site is based on the [Hugo](https://gohugo.io) system which is use in many other Eclipse projects as well.
It requires the hugo binary to be installed for generating the static HTML pages from the source code.
See README.md for details how to build the website locally. I have tried to include all/most of the content from the existing website.

boaks commented:
I'm right, that the README.md is a general one, which includes plenty of possible ways to install?
Trying just apt install hugo results in a too old version. 
Any recommendation, which of the approaches for installing hugo works most easily?

sophokles73 commented:
@boaks simply download the latest version from https://github.com/gohugoio/hugo/releases

If you ar on Ubuntu you might also want to try using the Snap
```
snap install hugo
```

boaks commented:
Thanks. I managed to install hugo and run the page with "hugo server -D".
Great work!
You removed the "copper" link. Is that by intention?

sophokles73 commented:
> You removed the "copper" link. Is that by intention?

Yes, because you marked it as *hibernated*, I was wondering if we still want it to be on the page. It is very little work to add it. I just wanted to make sure that we think it has value, given that it basically doesn't work anymore in any modern browser, or am I mistaken?

sbernard31 commented on site/content/community/channels/mailing-list.md:
```diff
@@ -0,0 +1,12 @@
+
+An excellent way to get the latest news and announcements around Californium is to join the mailing list.
+
+This is the place where we announce new releases and discuss architectural issues of greater impact.
```
I'm not sure the mailing is so much used ?

sophokles73 replied:
Would you like to change the wording only or do you want to remove the mailing list altogether?

sbernard31 replied:
My idea : if nobody use it, we can remove it.

But If you think, we must keep it. It's OK for me too.

boaks replied:
Yes, it's really rare used.
I would keep it, for those, who don't have a github account (even if they area also rare :-) .
Explaining, that github issues (or eclipse tickets) are usually better for focus on discussing a topic.

sophokles73 commented on site/content/_index/hero.md:
```diff
@@ -0,0 +1,34 @@
+background = "light" # can influence the text color
+particles = false
+
+title = "Hello"
```
If we want to keep the backdrop image then the title is ignored (and can be removed). If you would rather remove the backdrop then the title is used if I am not mistaken

sophokles73 commented on site/content/community/channels/index.md:
```diff
@@ -0,0 +1,8 @@
+title = "Get in Touch"
+subtitle = "Wanna tell us what you think about Eclipse Californium&trade;? What we can improve? Which feature is missing? Or just discuss architectural issues?"
+background = "dark"
++++
```
We can also include it in the main page. No problem. For the time being, there is not much content anyway so having everything on a single page might indeed make more sense.

sbernard31 replied:
:+1:

sophokles73 commented on site/content/_index/hero.md:
```diff
@@ -0,0 +1,34 @@
+#subtitle = "Eclipse Californium&trade;"
+
+[header]
+  image = "java.jpg"
```
@boaks what do you think? With or without the backdrop?

boaks replied:
no preference.

sophokles73 commented:
@sbernard31 I have moved the community section to the landing page. We could simply drop the first paragraph of the mailing description, would that work for you?

I have also removed the backdrop. Which background color do you like better, *light* or *secondardy*?

boaks commented:
The top field grey with only the Californium logo makes me feel, that there is something missing.
Is it possible to move the logo on field down and drop that grey?

boaks commented:
For me the boxes (with Californium, Actinium, Scandium) are optical more attractive, than the logo on grey on to much space :-).
But anyway, I think, go for it, and improve it ongoing.

sophokles73 commented:
> If someone has just the logo, I bet this should looks better. (more minimalist)

At least I don't have them. I'd say we start with the exiting *boxes* and once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

sophokles73 commented:
> The top field grey with only the Californium logo makes me feel, that there is something missing.

Are you talking about the header which includes the nav bar and the GitHub link at the right?

sbernard31 commented:
> once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

Don't challenge me or I will do it and we will finish with ugly logo instead of old school boxes. :grin:

sophokles73 commented:
@boaks 
I have replaced the logo in the header with the project name. Does that look better to you?

@sbernard31 
I have changed the wording for the mailing list. WDYT?

boaks commented:
> I have replaced the logo in the header with the project name. Does that look better to you?

yes.

(Even if I had the large grey box with the large californium logo in mind. But, please go ahead and merge it.)

boaks commented:
> once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

Sorry for the delay:
The logos are in the current webpage-repo, folder "dev/app/images/logos".
Cf, Ac, Sc, different sizes

![Ac_512](https://user-images.githubusercontent.com/10937819/68213823-93daf880-ffdc-11e9-9979-31afab54ceac.png)
![Cf_512](https://user-images.githubusercontent.com/10937819/68213853-9f2e2400-ffdc-11e9-93b7-3bb6dab714a9.png)
![Sc_512](https://user-images.githubusercontent.com/10937819/68213879-aa814f80-ffdc-11e9-9a97-3af60db5dc51.png)

sophokles73 commented:
I'll give it a shot then using the logos instead of the *dreaded* boxes ...

sophokles73 commented:
@sbernard31 @boaks 
I have pushed the updated site using the logos instead of the boxes. WDYT?

boaks commented:
Great. I would prefer the project name on the upper left corner instead of the logo, as it was before the last change.

sbernard31 commented:
> WDYT?

It seems better like this to me.
(but no problem if you prefer keep the boxes)

sophokles73 commented:
@boaks looks like we're getting close now, I have changed the nav bar according to your preference ...

boaks commented:
LGTM
